compiler/rustc_session/src/config.rs

```diff
@@ -2909,7 +2901,7 @@ pub fn build_session_options(early_dcx: &mut EarlyDiagCtxt, matches: &getopts::M
         describe_lints,
         output_types,
         search_paths,
-        maybe_sysroot: sysroot_opt,
+        maybe_sysroot: Some(sysroot),
```

Comment:
This made it impossible for the `config` driver callback to determine whether the `--sysroot` flag was passed or not, which is a problem for drivers that want to overwrite the default sysroot but honor explicit `--sysroot`. I'm trying to fix this in https://github.com/rust-lang/rust/pull/122993 but I don't really know what I am doing...